app/post_metadata_test.go

```diff
@@ -235,6 +235,48 @@ func TestPreparePostForClient(t *testing.T) {
 		})
 	})
 
+	t.Run("emojis overriding profile icon", func(t *testing.T) {
+		th := setup()
+		defer th.TearDown()
+
+		th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.ServiceSettings.EnablePostIconOverride = false
+		})
+
+		post, err := th.App.CreatePost(&model.Post{
+			UserId:    th.BasicUser.Id,
+			ChannelId: th.BasicChannel.Id,
+			Message:   "Test",
+		}, th.BasicChannel, false)
+
+		require.Nil(t, err)
+
+		post.AddProp(model.POST_PROPS_OVERRIDE_ICON_URL, "http://host.com/image.png")
+		post.AddProp(model.POST_PROPS_OVERRIDE_ICON_EMOJI, "basketball")
+
+		clientPost := th.App.PreparePostForClient(post, false, false)
+
+		t.Run("does not override icon URL", func(t *testing.T) {
```

Comment:
nit: this is probably just copy/paste from other tests, but I wonder if we could make each subtest more standalone vs. just subtesting the part of the code that asserts?

Reply:
Not sure if how I changed it now is what you had in mind. If not, please let me know.